src/charts/AreaChart/AreaChart.tsx

```diff
@@ -23,50 +33,114 @@ export default function AreaChart({
   colorScheme = d3.schemeCategory10
 }:AreaChartProps<string | number>):JSX.Element {
 
-  const anchor = useRef(null as unknown as SVGSVGElement);
-  const [windowSize, setWindowSize] = useState<[number, number]>([0, 0]);
-  const [cHeight, setCHeight] = useState<number>(0);
-  const [cWidth, setCWidth] = useState<number>(0);
+  const chart = 'AreaChart';
+
+  const {anchor, cHeight, cWidth}  = useResponsive();
+  const margin = useMemo(
+    () => getMargins(xAxis, yAxis, xAxisLabel, yAxisLabel),
+    [xAxis, yAxis, xAxisLabel, yAxisLabel]
+  )
+  const { xAxisX, xAxisY } = useMemo(
+    () => getXAxisCoordinates(xAxis, cHeight, margin),
+    [cHeight, xAxis, margin]
+  )
+  const { yAxisX, yAxisY } = useMemo(
+    () => getYAxisCoordinates(yAxis, cWidth, margin),
+    [cWidth, yAxis, margin]
+  )
+  // offset group to match position of axes
+  const translate = `translate(${margin.left}, ${margin.top})`
+  
+  // type KeyType = { key: string; dataType?: "number" | "date" | undefined; }
+  
+  // if no xKey datatype is passed in, determine if it's Date
+  if (!xDataType) {
+    xDataType = inferXDataType(data[0], xKey);
+  }
 
-  function updateSize() {
-    setWindowSize([window.innerWidth, window.innerHeight]);
+  // generate arr of keys. these are used to render discrete areas to be displayed
+  const keys: string[] = [];
+  if (groupBy) {
+    for (const entry of data) {
+      const property = String(entry[groupBy ?? ""]);
+      if (property && !keys.includes(property)) {
+        keys.push(property)
+      }
+    }
+    data = transformSkinnyToWide(data, keys, groupBy, xKey, yKey);
+  } else {
+    keys.push(yKey)
   }
 
-  // Set up an event listener on mount
-  useEnvEffect(() => {
-    window.addEventListener("resize", updateSize);
-    updateSize();
-    return () => window.removeEventListener("resize", updateSize);
-  }, []);
+  // generate stack: an array of Series representing the x and associated y0 & y1 values for each area
+  const stack = d3.stack().keys(keys)
+  const layers = useMemo(() => {
+    const layersTemp = stack(data as Iterable<{ [key: string]: number; }>);
+    for (const series of layersTemp) {
+      series.sort((a, b) => b.data[xKey] - a.data[xKey]);
+    }
+    return layersTemp;
+  }, [data, keys])
+  
+  const xAccessor: xAccessorFunc = xDataType === 'number' ? (d) => d[xKey] : (d) => new Date(d[xKey]);
+  const yAccessor: yAccessorFunc = (d) => d[yKey];
+  
+  const {xScale, xMin, xMax} = xScaleDef(data, xDataType, xAccessor, margin, cWidth, chart);
+  const yScale = yScaleDef(layers, yAccessor, margin, cHeight, chart);   
```

Comment:
Not sure why this makes sense passing layers in to yScaleDef and data into xScaleDef, but can't argue with results. Would've thought it would be data into both.